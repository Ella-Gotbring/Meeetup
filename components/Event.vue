<template>
  <div class="hello">
    <hr />
    <p>Register to a meetup today!</p>
    <hr />
    <div>
      <ul class="row">
        <li v-for="item in list" :key="item.id">
          <div class="column">
            <div class="location">
              {{ item.location }}
            </div>
            <div class="date">
              {{ item.date }}
            </div>

            <p class="reg">
              <router-link class="button" to="/Register" tag="button"
                >Register</router-link
              >
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <!-- </div> -->
</template>

<script>
import { getLocation } from "../data/data.js";

export default {
  name: "Event",

  data: () => ({
    list: [],
  }),
  created() {
    this.list = getLocation();
  },
  // methods: {
  //   register: function(message) {
  //     alert(message);
  //   },
  // },
};
</script>

<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
.location {
  font-size: 1.5rem;
  margin: 20px;
  padding: 10px;
}
.date {
  font-size: 2rem;
}
</style>
