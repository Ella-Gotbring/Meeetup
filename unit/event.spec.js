import { shallowMount } from "@vue/test-utils";
import Event from "@/components/Event.vue";

describe("Event.vue", () => {
  test("events are being showed", () => {
    const wrapper = shallowMount(Event);
    expect(wrapper).toMatchSnapshot();

    //events amount to be 3

    expect(wrapper);
  });
});
