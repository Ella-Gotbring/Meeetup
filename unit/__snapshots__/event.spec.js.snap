// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event.vue events are being showed 1`] = `
<div class="hello">
  <hr>
  <p>Register to a meetup today!</p>
  <hr>
  <div>
    <ul class="row">
      <li>
        <div class="column">
          <div class="location">
            Stockholm
          </div>
          <div class="date">
            5/10
          </div>
          <p class="reg">
            <router-link to="/Register" tag="button" class="button">Register</router-link>
          </p>
        </div>
      </li>
      <li>
        <div class="column">
          <div class="location">
            Milano
          </div>
          <div class="date">
            20/10
          </div>
          <p class="reg">
            <router-link to="/Register" tag="button" class="button">Register</router-link>
          </p>
        </div>
      </li>
      <li>
        <div class="column">
          <div class="location">
            London
          </div>
          <div class="date">
            18/11
          </div>
          <p class="reg">
            <router-link to="/Register" tag="button" class="button">Register</router-link>
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
